### PR TITLE
[#IOCOM-272] Integrated Lollipop with Remote Content 

### PR DESCRIPTION
**Pull request by Garma00**
#### List of Changes

* [Update third party openati to openapi3](https://github.com/pagopa/io-backend/commit/0932dbb9e05a673e5c840eb0afa52832d3fc88ca);
* [Update codegen version](https://github.com/pagopa/io-backend/commit/23253db0cfc420ed40c3cd7787602653cf9a05e7);
* [Move lollipop definition outside of generated](https://github.com/pagopa/io-backend/commit/e34843e53c2725762783c0b60933a4b03e843838);
* [Add broken codegen yaml file](https://github.com/pagopa/io-backend/commit/549f79c6151b045627ee77cf015f12cdad8bead2);
* [Add lollipop headers to api-piattaforma-notifiche](https://github.com/pagopa/io-backend/commit/a2c97862e7bdf8e721f141b5a5c3906bbeaada06);

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by michaeldisaro — CHANGES_REQUESTED** [review]

File: src/services/__tests__/newMessagesService.test.ts [hunk 1]
```diff
@@ -98,20 +118,29 @@ const validApiThirdPartyMessageResponse = {
...
-      }
-    }
-  }
+        third_party_data: { id: aValidThirdPartyMessageUniqueId },
+      },
+    },
+  },
 };
 
+const MessageWithThirdPartyData = t.intersection([
```
[the remark this answers is not in the document]
Comment (Garma00) [reply src/services/__tests__/newMessagesService.test.ts]: Did in [Export MessageWithPartyData without redefining it into the test file](https://github.com/pagopa/io-backend/pull/1021/commits/c977fcb04a9a7bfe23af37fb0589e7a647b00364)

File: src/controllers/__tests__/messagesController.test.ts [hunk 2]
```diff
@@ -339,32 +396,43 @@ describe("MessagesController#getThirdPartyAttachment", () => {
   it("should call the getThirdPartyAttachment on the messagesController with valid values", async () => {
     const req = mockReq();
 
+    // TODO: Define a const with aMessageWithThirdPartyData with values
```
[the remark this answers is not in the document]
Comment (Garma00) [reply src/controllers/__tests__/messagesController.test.ts]: Did in [Remove TODO](https://github.com/pagopa/io-backend/pull/1021/commits/80034ad66c058591694f647f849cb1c60c3f4837)

File: src/clients/__tests__/third-party-service-client.test.ts [hunk 3]
```diff
@@ -141,20 +149,18 @@ describe("third-party-service-client", () => {
...
     });
     const expectedConfig = aValidTestAndProdThirdPartyConfig.prodEnvironment!;
 
-    expect(res).toMatchObject(
-      E.right({
-        status: 200,
-        value: aPNThirdPartyNotification
-      })
-    );
+    console.log(await mockNodeFetch())
```
[the remark this answers is not in the document]
Comment (Garma00) [reply src/clients/__tests__/third-party-service-client.test.ts]: did in [Remove console.log inside test](https://github.com/pagopa/io-backend/pull/1021/commits/0d773f3ef5913aff8eb3a411b64390d6929a96e7)

File: package.json [hunk 4]
```diff
@@ -60,7 +60,7 @@
     "generate:proxy:piattaforma-notifiche-courtesy": "rimraf generated/piattaforma-notifiche-courtesy && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pn-user-attributes/13b91a353f6a219c60d5cdadc023cc18c4157554/docs/openapi/pn-address-book-b2b-external-io-v1.yaml --out-dir generated/piattaforma-notifiche-courtesy --request-types --response-decoders --client",
     "generate:test-certs": "./scripts/generate-test-certs.sh certs",
     "generate:proxy:api-session": "rimraf generated/session && gen-api-models --api-spec api_session.yaml --out-dir generated/session",
-    "generate:lollipop-definitions": "rimraf generated/lollipop && gen-api-models --api-spec openapi/generated/lollipop_definitions.yaml --out-dir generated/lollipop",
+    "generate:lollipop-definitions": "rimraf generated/lollipop && gen-api-models --api-spec openapi/lollipop_definitions.yaml --out-dir generated/lollipop",
```
[the remark this answers is not in the document]
Comment (Garma00) [reply package.json]: Did in [Move back lollipop_definitions](https://github.com/pagopa/io-backend/pull/1021/commits/47a3c644d8d9c55b3aec376945408ad82067c904)

File: src/controllers/__tests__/messagesController.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaeldisaro) [reply src/controllers/__tests__/messagesController.test.ts]: I left the comment not to just remove the todo, but to do it :)

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (michaeldisaro) [reply package.json]: Sorry I did not meant this. I was misleaded by another change that we made locally. It's correct that lollipop.definitions are pulled out from "generated" folder. You can revert this. Sorry.

[the remark this answers is not in the document]
Comment (Garma00) [reply package.json:63]: [Revert "Move back lollipop_definitions"](https://github.com/pagopa/io-backend/pull/1021/commits/809c6874e0b33410e6828fcb42a8b6aa2c9127ec)

File: src/adapters/pnFetch.ts [hunk 5]
```diff
@@ -325,35 +340,73 @@ export const redirectAttachment =
...
-                )
-              ),
-              TE.mapLeft(errorResponse),
-              TE.toUnion
+    pipe(
+      maybeLollipopLocals,
+      O.fromNullable,
+      O.fold(
+        () =>
+          Promise.resolve(
```
[the remark this answers is not in the document]
Comment (Garma00) [reply src/adapters/pnFetch.ts]: Ok pls take a look here @michaeldisaro @acuiuli 😄  [Make lollipop params optional for PN fetch](https://github.com/pagopa/io-backend/pull/1021/commits/1e2f3a82c6abf664af6c7fcdbcce6671d1aa1bfb)

File: src/controllers/__tests__/messagesController.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Garma00) [reply src/controllers/__tests__/messagesController.test.ts]: [Add a message response in test instead of an empty object](https://github.com/pagopa/io-backend/pull/1021/commits/7a858812bec479e40dc57b3ab2749cbf362574d6)

File: src/controllers/__tests__/messagesController.test.ts [hunk 6]
```diff
@@ -339,32 +396,43 @@ describe("MessagesController#getThirdPartyAttachment", () => {
...
 
+    console.log(lollipopRequiredHeaders);
+
     const response = await controller.getThirdPartyMessageAttachment(req);
 
     expect(mockGetThirdPartyAttachment).toHaveBeenCalledWith(
-      mockedUser.fiscal_code,
-      req.params.id,
-      req.params.attachment_url
+      {},
```
[the remark this answers is not in the document]
Comment (Garma00) [reply src/controllers/__tests__/messagesController.test.ts]: [Add a comment to explain the undefined lollipop locals in test](https://github.com/pagopa/io-backend/pull/1021/commits/490d7c190a94fc6e58e96763700b1ae0659819e2)

**Review by michaeldisaro — APPROVED** [review]
LGTM
